### PR TITLE
Description should be a textarea

### LINKED ISSUE #7970: Description should be a textarea
### Description

We can change make the description as a text area on both the Milestone and Events form pages instead of a text box. Please assign it to me and I can fix it.

### Screenshots

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sumitunique777**
## Fixes Issue 

fixes #7970

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8218"></a>

### CODE REVIEW

File: components/form/Textarea.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+              ? "bg-primary-low-medium dark:bg-primary-medium-low hover:border-primary-medium-low focus:ring-0 focus:border-primary-medium focus:outline-0 cursor-not-allowed"
+              : "dark:bg-primary-high hover:border-tertiary-medium focus:ring-0 focus:border-tertiary-medium focus:outline-0"
+          }"}`}
+          id={name}
+          name={name}
+          {...restProps}
+        />
+      </>
+    );
+  }
```
Comment (eddiejaoude) [inline components/form/Textarea.js:29]: Blank line ending required...

```suggestion
}

```

File: components/user/UserProfile.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ function UserProfile({ BASE_URL, data }) {
           </div>
         </div>
       </div>
-      <div className="flex justify-center my-4 text-center">
+      <div className="justify-center my-4 text-center">
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:71]: This still shows multiple lines from the textarea as 1 line

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you 👍  I left an inline comment

**Comment by eddiejaoude** [comment]
Previous conversation in an older PR https://github.com/EddieHubCommunity/LinkFree/pull/8060

@sumitunique777 please continue to use the same PR

File: components/user/UserProfile.js (hunk 2 above)
Comment (sumitunique777) [reply components/user/UserProfile.js]: I have committed changes for this issue

**Comment by sumitunique777** [comment]
I have committed changes

File: components/user/UserProfile.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ function UserProfile({ BASE_URL, data }) {
   const LinkRenderer = ({ href, children }) => (
     <Link href={href}>{children}</Link>
   );
-
+
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:36]: This change is not required

```suggestion

```

File: components/user/UserProfile.js [hunk 4]
```diff
@@ -68,10 +67,10 @@ function UserProfile({ BASE_URL, data }) {
...
         </div>
       </div>
-      <div className="flex justify-center my-4 text-center">
-        <ReactMarkdown components={{ a: LinkRenderer }}>
-          {data.bio}
-        </ReactMarkdown>
+      <div className="justify-center my-4 text-center whitespace-pre-wrap">
+  <ReactMarkdown  components={{ a: LinkRenderer }}>
+         {data.bio.replace(/\n/gi, '\n &nbsp;')}
+</ReactMarkdown>
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:73]: Formatting is not correct here, using the VScode Prettier plugin with format this automatically for you

```suggestion
        <ReactMarkdown components={{ a: LinkRenderer }}>
          {data.bio.replace(/\n/gi, "\n &nbsp;")}
        </ReactMarkdown>
```

File: components/user/UserProfile.js [hunk 5]
```diff
@@ -68,9 +67,9 @@ function UserProfile({ BASE_URL, data }) {
           </div>
         </div>
       </div>
-      <div className="flex justify-center my-4 text-center">
+      <div className="justify-center my-4 text-center whitespace-pre-wrap">
         <ReactMarkdown components={{ a: LinkRenderer }}>
-          {data.bio}
+          {data.bio.replace(/\n/gi, "\n &nbsp;")}
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:72]: Should this happen in the markdown plugin, so all output from text area can benefit from multiple lines?

Comment (eddiejaoude) [reply components/user/UserProfile.js:72]: This worked well for the Profile...

Screenshot 2023-07-23 at 15 14 58

But other textareas will need this functionality also, for example events has a textarea but the output does not obey the same line breaks ...

Screenshot 2023-07-23 at 15 15 39

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good, almost there 👍 

Thank you, I left some inline comments

**Comment by sumitunique777** [comment]
I have committed some changes regarding this issue for now I haven't done any changes in the ReactMarkDown plugin

**Comment by sumitunique777** [comment]
Will you please review my changes that I have committed

**Comment by eddiejaoude** [comment]
> Will you please review my changes that I have committed

But you have not addressed all the [comments](https://github.com/EddieHubCommunity/LinkFree/pull/8218#discussion_r1271455354), and you have conflicts?

File: components/user/UserProfile.js (hunk 5 above)
Comment (sumitunique777) [reply components/user/UserProfile.js:72]: > Should this happen in the markdown plugin, so all output from text area can benefit from multiple lines?

Will you please help me in this? how can I make changes in plugin code to make the things work

**Comment by eddiejaoude** [comment]
Please can you fix the failing test

File: components/event/EventCard.js [hunk 6]
```diff
@@ -84,8 +84,8 @@ export default function EventCard({ manage, event, usernames }) {
                 <MdOutlineArrowRightAlt className="self-center hidden lg:block" />
                 <span>{endTime}</span>
               </p>
-              <ReactMarkdown className="text-sm text-primary-medium dark:text-primary-low-medium py-1 flex-wrap">
-                {event.description}
+              <ReactMarkdown className="text-sm text-primary-medium dark:text-primary-low-medium py-1 flex-wrap whitespace-pre-wrap ">
+                {event.description.replace(/\n/gi, '\n &nbsp;')}
```
Comment (azhechkova) [inline components/event/EventCard.js:88]: Duplicate logic in UserProfile component. Maybe it's better to move it to utils

Comment (eddiejaoude) [reply components/event/EventCard.js:88]: The markdown library we are using allows plugins, which looks like there is a plugin for line breaks from the user

- List of plugins https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins
- Line break plugin https://github.com/remarkjs/remark-breaks

Comment (sumitunique777) [reply components/event/EventCard.js:88]: Should I proceed with the previous logic i have used ?

Comment (eddiejaoude) [reply components/event/EventCard.js:88]: No. I will have a play with those plugins

**Comment by eddiejaoude** [comment]
Thank you for your Pull Request. While having a play I found a solution that didn't require regex or plugins, you can see it here https://github.com/EddieHubCommunity/LinkFree/pull/8594 - so I will close this PR.

**Pull request closed without merging**